Packs/IOCParser/Integrations/IOCParser/IOCParser.py

```diff
@@ -0,0 +1,417 @@
+        """
+        Extracts all IOCs from a URL
+
+        :type url: ``str``
+        :param url: The URL from which the IOCs will be extracted
+
+        :return: The HTTP response returned by the API
+        :rtype: ``Dict[str, Any]``
+        """
+
+        data = {'url': url, 'public': False}
+        return self._http_request(
+            method='POST',
+            url_suffix='/url',
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps(data),
+            return_empty_response=True
+        )
+
+    def ioc_from_text(self, text: str) -> Dict[str, Any]:
+        """
+        Extracts all IOCs from a JSON text
+
+        :type text: ``str``
+        :param text: The JSON from which the IOCs will be extracted
+
+        :return: The HTTP response returned by the API
+        :rtype: ``Dict[str, Any]``
+        """
+        data = {'data': text}
+        return self._http_request(
+            method='POST',
+            url_suffix='/text',
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps(data),
+            return_empty_response=True
+        )
+
+    # TODO the api gets raw text not as dict
+    def ioc_from_raw_text(self, raw_text: str, keys: list = None) -> Dict[str, Any]:
+        data = {}
+        if keys:
+            data['keys'] = keys
+        return self._http_request(
+            method='POST',
+            url_suffix='/raw',
+            headers={'Content-Type': 'text/plain'},
+            data=json.dumps(data),
+            return_empty_response=True
+        )
+
+    def ioc_from_twitter(self, user_name: str) -> Dict[str, Any]:
+        """
+        Extracts all IOCs from a twitter account
+
+        :type user_name: ``str``
+        :param user_name: The twitter account from which the IOCs will be extracted
+
+        :return: The HTTP response returned by the API
+        :rtype: ``Dict[str, Any]``
+        """
+        data = {'data': user_name}
+        return self._http_request(
+            method='POST',
+            url_suffix='/twitter',
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps(data),
+            return_empty_response=True
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def list_to_upper_case(lst: List[str]) -> List[str]:
+    """
+    Upper case every string in the list
+
+    :type lst: ``List``
+    :param lst: The list we want to upper case
+
+    :return: The upper cased list
+    :rtype: ``List``
+    """
+    return list(map(lambda x: x.upper(), lst))
+
+
+def remove_unwanted_keys(response_data: Dict[str, List], keys: List[str]) -> None:
+    """
+    Removes all keys that were not specified by the user as the desired keys to return
+
+    :type response_data: ``Dict[str, List]``
+    :param response_data: The data key from the API's response
+
+    :type keys: ``List[str]``
+    :param keys: IOC Types to return
+    """
+
+    keys_list = list(response_data.keys())
```

Comment:
```suggestion
    keys_list = response_data.keys()
```

You can still iterate through it.